src/game/puppyprint.c

```diff
@@ -878,148 +1182,269 @@ void print_small_text(s32 x, s32 y, const char *str, s32 align, s32 amount, s32
-
-        case '~': *textX = 40; *textY = 24; *spaceX = textLen[74]; break; // Tilde
-        case '@': *textX = 44; *textY = 24; *spaceX = textLen[75]; break; // Umlaut
-        case '^': *textX = 48; *textY = 24; *spaceX = textLen[76]; break; // Caret
-        case '/': *textX = 52; *textY = 24; *spaceX = textLen[77]; break; // Slash
-        case '_': *textX = 56; *textY = 24; *spaceX = textLen[78]; break; // Percent
-        case '&': *textX = 60; *textY = 24; *spaceX = textLen[79]; break; // Ampersand
-
-        // This is for the letters that sit differently on the line. It just moves them down a bit.
-        case 'g': *offsetY = 1; break;
-        case 'q': *offsetY = 1; break;
-        case 'p': if (font == FONT_DEFAULT) *offsetY = 3; break;
-        case 'y': if (font == FONT_DEFAULT) *offsetY = 1; break;
+u8 gIsLightText = FALSE;
+
+// This is where the deferred printing will be stored. When text is made, it will store text with an 12 byte header, then the rest will be the text data itself.
+// The first 4 bytes of the header will be the X and Y pos
+// The next 4 bytes will be the current envcolour set by print_set_envcolour
+// Then the string length, text alignment, amount and font each get a byte.
+// The data afterwards is the text data itself, using the string length byte to know when to stop.
+#define HEADERSIZE 13
+void print_small_text_buffered(s32 x, s32 y, const char *str, u8 align, s32 amount, u8 font) {
+    u8 strLen = MIN((signed)strlen(str), 255);
+    // Compare the cursor position and the string length, plus 12 (header size) and return if it overflows.
+    if (sPuppyprintTextBufferPos + strLen + HEADERSIZE > sizeof(sPuppyprintTextBuffer))
+        return;
+    x += 2048;
+    y += 2048;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 0] = (x >> 8) & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 1] = (x & 0xFF);
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 2] = (y >> 8) & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 3] = (y & 0xFF);
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 4] = gCurrEnvCol[0];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 5] = gCurrEnvCol[1];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 6] = gCurrEnvCol[2];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 7] = gCurrEnvCol[3];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 8] = strLen;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 9] = align;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 10] = (amount == -1) ? 255 : amount;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 11] = font;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 12] = gIsLightText;
+    bcopy(str, &sPuppyprintTextBuffer[sPuppyprintTextBufferPos + HEADERSIZE], strLen);
+    sPuppyprintTextBufferPos += strLen + HEADERSIZE;
+}
+
+void print_small_text_buffered_light(s32 x, s32 y, const char *str, u8 align, s32 amount, u8 font) {
+    gIsLightText = TRUE;
```

Comment:
Is this variable only used internally or should the value of `gIsLightText` be set to a temporary variable here?

Reply:
Just an internal tracker, yes. Var has been made static to reflect that.